app/Traits/ArclightConnection.php

```diff
@@ -25,9 +25,16 @@ private function fetchArclight($path, $language_id = null, $include_refs = false
         $path .= '&'.$parameters;
 
         try {
-            $results = json_decode(file_get_contents($path));
+            $ctx = stream_context_create(array('http'=>
+                array('timeout' => 5)
+            ));       
+            $results = json_decode(file_get_contents($path, false, $ctx));
         } catch (Exception $e) {
-            return null;
+            $error = 
+                strpos($e, 'timed out') !== false ? 
+                $this->setStatusCode(408)->replyWithError('Request timeout') :
+                $this->setStatusCode(500)->replyWithError('Internal server error');
```

Comment:
If we are going to return 500 Internal Server Error, we should log as much info as we can so that we can fix the issue. HTTP 500 errors are a personal affront to me as an API developer - they should be hunted down and killed.